src/main/java/com/azure/cosmos/kafka/connect/sink/CosmosDBSinkTask.java

```diff
@@ -83,8 +91,15 @@ public void put(Collection<SinkRecord> records) {
 
                 try {
                     addItemToContainer(container, recordValue);
-                } catch (BadRequestException bre) {
-                    throw new CosmosDBWriteException(record, bre);
+                } catch (CosmosException | ConnectException bre) {
+                    if (config.getString(TOLERANCE_ON_ERROR_CONFIG).equalsIgnoreCase("all")) {
+                        logger.error("Could not upload record to CosmosDb, but tolerance is set to all. Value: {}."
+                                + " Error: {}", recordValue.toString(), bre.getMessage());
+                    } else {

```

Comment:
is this why you added a new config item? 
by default, if you set ```errors.tolerance = all``` Kafka Connect will ignore the message causing the error and proceed, but it doesn't log anything. 

Could we just check the value of ```errors.tolerance``` here instead of a new specific config?